### DISCUSSION TITLE
gauge_custom_compile_dir with gradle gauge plugin

### DISCUSSION THREAD

**Discussion by balajiprabuk** [getgauge/gauge#2185] [General]
Question: How do we set "gauge_custom_compile_dir", when using gradle gauge plugin.?

I am able to run gauge command directly with the "gauge_custom_compile_dir" value set in my env/default/java.properties and gauge rightly picks up .java files mentioned in "gauge_custom_compile_dir".

But when i run gradle gauge, it is not picking up "gauge_custom_compile_dir" value from env/default/java.properties. It is defaulting to usual test folder in gradle project structure.

**Comment by zabil** [comment]
Gradle projects use gradle specific defaults for build and test output directories. Have you tried setting the [`buildDir`](https://docs.gradle.org/current/userguide/writing_build_scripts.html#sec:standard_project_properties) to a custom value?

**Comment by balajiprabuk** [comment]
We have gradle project, 
1. we have unit tests written with @Test annotation.
2. In this same project, we also have the acceptance tests written in gauge.

When we run ./gradlew gauge, this triggers the unit test as well only after which the gauge tests are running.
We dont want the unit tests to run when we run the gradle gauge command.

Can you suggest recommended approach here?

**Reply by zabil** [reply]
Have you tried [skipping test](https://www.baeldung.com/gradle-skip-tests) flag?

**Reply by balajiprabuk** [reply]
We ran with ./gradlew gauge -x test, but this still runs all the unit tests

**Reply by sriv-tw** [reply]
It seems like this is not possible since the gauge-gradle-plugin extends the gradle `Test` task and I guess that is what's causing it to inherit the behaviour too. It will require a change in the gauge-gradle-plugin to avoid this. The change to extend `Test` was required to support sourceSets, but I don't know if that can be used in your case to isolate between tests.

**Reply by balajiprabuk** [reply]
We just found a way around this. 

Initially our @Test annotations were imported from `import org.junit.Test;`, but after changing it jupiter annotation with `import org.junit.jupiter.api.Test;`, 

./gradlew gauge command by itself does not run the unit tests and just works as expected.
